internal/provider/data_source_gitlab_group_membership_test.go

```diff
@@ -89,3 +111,54 @@ data "gitlab_group_membership" "foomaintainers" {
   access_level = "maintainer"
 }`, rInt, rInt)
 }
+
+func testAccDataSourceGitlabGroupMembershipPagination(rInt int, userCount int) string {
+	t, err := template.New("users").Parse(fmt.Sprintf(`
```

Comment:
Yes, there is definitely room for improvement there :D 

About the `count` - I spent too much time trying to get that to work. The problem is that `count` can be used fine to create the users, but for the group membership you need the user ids, which are only known during apply and you would theoretical have to use `for_each` - but yeah, doesn't work if it's only known during apply with `for_each` 🤷 

I'd love to learn about a creative proper solution :D

Reply:
Two ideas I have 
1. Do it in two TestSteps, where the first one creates the resources and the second one adds the datasource

so the first apply configuration could look like:

```terraform
resource "gitlab_group" "this" {
  name = "foo-%[1]d"
  path = "foo-%[1]d"
}

resource "gitlab_user" "this" {
  count    = %[2]d
  name     = "foo-%[1]d-${count.index}"
  username = "foo-%[1]d-${count.index}"
  password = "foo"
  email    = "foo-%[1]d-${count.index}@example.com"
}

resource "gitlab_group_membership" "this" {
  count        = %[2]d
  group_id     = gitlab_group.this.id
  user_id      = gitlab_user.this[count.index].id
  access_level = "developer"
}
```

where [1] is the randomid and [2] is 21

2. Create the groups, users, and memberships outside of the TestStep using the gitlab client directly, and then only include the `gitlab_group_membership` datasource in the apply config (no other resources)

Reply:
I like option 2 because IMO we should only include TF config for the thing being tested. It keeps tests isolated, is faster, and lets us correctly check the destroy functions using the `CheckDestroy` function. (Related to #205)

Reply:
Nice!